corehq/apps/reports/analytics/esaccessors.py

```diff
@@ -193,3 +193,69 @@ def get_form_counts_by_user_xmlns(domain, startdate, enddate, user_ids=None,
                 counts[key] = xmlns_bucket.doc_count
 
     return counts
+
+
+def get_form_duration_stats_by_user(
+        domain,
+        app_id,
+        xmlns,
+        user_ids,
+        startdate,
+        enddate,
+        by_submission_time=True):
+    """Gets stats on the duration of a selected form grouped by users"""
+    date_filter_fn = submitted_filter if by_submission_time else completed_filter
+
+    query = (
+        FormES()
+        .domain(domain)
+        .app(app_id)
+        .user_id(user_ids)
+        .xmlns(xmlns)
+        .filter(date_filter_fn(gte=startdate, lt=enddate))
+        .aggregation(
+            TermsAggregation('user_id', 'form.meta.userID').aggregation(
+                ExtendedStatsAggregation(
+                    'duration_stats',
+                    'form.meta.timeStart',
+                    script="doc['form.meta.timeEnd'].value - doc['form.meta.timeStart'].value",
```

Comment:
The other issue here is that ideally we shouldn't use scripts since having them enabled on the cluster is a security risk. Unfortunately the PACT reports already use them so it is enabled on prod but not on any other environment. 

If we do go ahead with this we'd have to update the ES settings and restart ES on all the other environments. It's all set up in ansible, just need to do it. Doing an xform reindex takes about 50 hours on prod.